### PR TITLE
JavaScript: Do not require('assert')

### PR DESCRIPTION
**Pull request by glebm**
Removes require('assert'), as the current version causes compiler and linter warnings, and fixing the checks to be safe is not worth the effort since we only ever use the trivial `assert.ok`.

Enables strict mode for the file (`"use strict";`).

Compiles waxeye.coffee with the latest coffeescript (v2.0.1).

### CODE REVIEW

File: src/javascript/waxeye.js [hunk 1]
```diff
@@ -31,6 +26,7 @@
   };
 
   uniq = function(a) {
+    // Why sort here? Seems useful to keep these in the original order.
```
Comment (jishi9) [inline src/javascript/waxeye.js]: +1

File: src/javascript/waxeye.js [hunk 2]
```diff
@@ -379,22 +353,33 @@
...
                   }
+                  // JavaScript string comparison does not compare Unicode characters correctly, so we must compare
+                  // codepoints. Example:
+                  //   "ﬆ" > '𝌆' //=> true
+                  //   "ﬆ".codePointAt(0) > '𝌆'.codePointAt(0) //=> false
                   inputCodePoint = input.codePointAt(pos);
+                  // Loop over cc instead of recursing to avoid stack overflow on large character classes.
                   for (j = 0, len = cc.length; j < len; j++) {
                     charClass = cc[j];
+                    // Single character
```
Comment (jishi9) [inline src/javascript/waxeye.js]: Is something missing here? If not, the comments are misleading

File: src/javascript/waxeye.js [hunk 3]
```diff
@@ -379,22 +353,33 @@
...
                     isMatch = typeof charClass === 'number' ? charClass === inputCodePoint : charClass[0] <= inputCodePoint && charClass[1] >= inputCodePoint;
                     if (isMatch) {
                       return MachineState.INTER(MachineConfiguration.APPLY(k, isSingleCharCodepoint(inputCodePoint) ? Value.VAL(pos + 1, arrayPrepend(input[pos], asts), err) : Value.VAL(pos + 2, arrayPrepend(input[pos] + input[pos + 1], asts), err)));
                     }
                   }
                   return MachineState.INTER(MachineConfiguration.APPLY(k, Value.FAIL(updateError(err, pos, new ErrCC(cc)))));
                 case "SEQ":
+                  // a sequence is made up of a list of expressions
+                  // we traverse the list, making sure each expression succeeds
+                  // the rest of the string return by the expression is used
```
Comment (jishi9) [inline src/javascript/waxeye.js]: typo: the rest of the string *returned* by

File: src/javascript/waxeye.js (hunk 2 above)
Comment (glebm) [reply src/javascript/waxeye.js]: This is a generated file.

**Comment by glebm** [comment]
Note that this PR is superseded by #78, but it might be useful to merge this first anyway so that there is an intermediate commit that still has the old JavaScript parser but with the warning fixed.

**Comment by orlandohill** [comment]
Thanks!

Yes, doing an intermediate commit makes sense to me.

**Pull request merged**